packages/components/src/components/toggle-button/toggle-button.tsx

```diff
@@ -0,0 +1,158 @@
+/**
+ * @license
+ * Scale https://github.com/telekom/scale
+ *
+ * Copyright (c) 2021 Egor Kirpichev and contributors, Deutsche Telekom AG
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+import {
+  Component,
+  Prop,
+  h,
+  Host,
+  Element,
+  Event,
+  EventEmitter,
+} from '@stencil/core';
+import classNames from 'classnames';
+
+enum iconSizes {
+  xs = '12',
+  small = '16',
+  regular = '22',
+  large = '24',
+}
+
+let i = 0;
+
+@Component({
+  tag: 'scale-toggle-button',
+  styleUrl: 'toggle-button.css',
+  shadow: true,
+})
+export class ToggleButton {
+  @Element() hostElement: HTMLElement;
+  /** (optional) The size of the button */
+  @Prop() size?: 'large' | 'regular' | 'small' | 'xs' = 'large';
+  /** (optional) Button variant */
+  @Prop() variant?: 'primary' | 'secondary' = 'primary';
+  /** (optional) background color scheme of a selected button */
+  @Prop() colorScheme?: 'magenta' | 'black' = 'magenta';
+  /** (optional) If `true`, the button is disabled */
+  @Prop() disabled?: boolean = false;
+  /** (optional) If `true`, the button is selected */
+  @Prop({ mutable: true }) selected?: boolean = false;
+  /** (optional) Button type */
+  @Prop() iconOnly?: boolean = false;
+  /** (optional) Icon position related to the label */
+  @Prop({ reflect: true, mutable: true }) iconPosition: 'before' | 'after' =
+    'before';
+  /** (optional) set the border-radius left, right or both */
+  @Prop() radius: 'left' | 'right' | 'both' | null = null;
+  /** (optional) toggle button's id */
+  @Prop() toggleButtonId?: string;
+  /** (optional) aria-label attribute needed for icon-only buttons */
+  @Prop() ariaLabel: string;
+  /** (optional) Injected CSS styles */
+  @Prop() styles?: string;
+  /** Emitted when button is clicked */
+  @Event() scaleClick!: EventEmitter<{ id: string; selected: boolean }>;
+
+  connectedCallback() {
+    this.setIconPositionProp();
+  }
+
+  componentDidLoad() {
+    this.handleIconSize();
+  }
+
+  componentDidRender() {
+    this.handleIconSize();
+  }
+
+  componentWillLoad() {
+    if (this.toggleButtonId == null) {
+      this.toggleButtonId = 'toggle-button-' + i++;
+    }
+  }
+
+  handleIconSize() {
+    const icon = this.hostElement.children[0] && this.hostElement.children[0];
```

Comment:
what if `this.styles` is truthy? then the first child will be <styles>, right?

Reply:
what if `this.styles` is truthy? then the first child will be `<styles>...</styles>`, right?

Reply:
<styles>...</styles> will not be the first child, when being applied. I think the children are the slotted elements (text and/or icon and 'unwished' text nodes due tue formatting, which are eliminated).